File: solutions/csharp/bird-watcher/1/BirdWatcher.cs
using System;

class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek()
    {
        int[] visitedLastWeek = [0, 2, 5, 3, 7, 8, 4];
        return visitedLastWeek;
    }

    public int Today()
    {
        return birdsPerDay[6];
    }

    public void IncrementTodaysCount()
    {
        birdsPerDay[6] += 1;
    }

    public bool HasDayWithoutBirds()
    {
        foreach(int element in birdsPerDay) 
        {
            if (element == 0) 
            {
                return true;
            }
        }
        return false;
    }

    public int CountForFirstDays(int numberOfDays)
    {
        int counter = 0;
        for (int i = 0; i < numberOfDays; i++)
        {
            counter += birdsPerDay[i];
        }
        return counter;
    }

    public int BusyDays()
    {
        int busyDays = 0;
        
        foreach(int element in birdsPerDay) 
        {
            if (element >= 5) 
            {
                busyDays++;
            }
        }
        return busyDays;
    }
}

File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java
public final class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        return this.birdsPerDay;
    }

    public int getToday() {
        return this.birdsPerDay[6];
    }

    public void incrementTodaysCount() {
        this.birdsPerDay[6] += 1;
    }

    public boolean hasDayWithoutBirds() {
        boolean dayWithoutBirds = false;

        for (int count : this.birdsPerDay) {
            if (count == 0) {
                dayWithoutBirds = true;
            }
        }

        return dayWithoutBirds;
    }

    public int getCountForFirstDays(int numberOfDays) {
        int total = 0;

        if (numberOfDays > 7) {
            numberOfDays = 7;
        }

        for (int count = 0; count < numberOfDays; count++) {
            total += this.birdsPerDay[count];
        }

        return total;
    }

    public int getBusyDays() {
        int busyDays = 0;

        for (int count : this.birdsPerDay) {
            if (count >= 5) {
                busyDays += 1;
            }
        }

        return busyDays;
    }
}
